app/models/team.rb

```diff
@@ -26,4 +26,9 @@ def self.all_non_special
   def member_ids
     team_users.pluck(:user_id)
   end
+
+  # Returns all teams whose name match the query
+  def self.search_from_query(valid_teams, query)
+    where(id: valid_teams).where(hidden: 0).where(arel_table[:name].matches(query))
```

Comment:
Use the `all_non_special` method for the `where(hidden: 0)` part. Moreover, could you transform the `all_non_special` method into a scope ? (while preserving all its documentation of course).